example/glide_computer/include/glide_computer.h

```diff
@@ -56,60 +51,43 @@
 // - no ground obstacles (i.e. mountains) to pass
 // - flight path exactly on a shortest possible line to destination
 
-template<units::QuantityKind QK>
-struct STD_FMT::formatter<QK> : formatter<typename QK::quantity_type> {
-  template<typename FormatContext>
-  auto format(const QK& v, FormatContext& ctx)
-  {
-    return formatter<typename QK::quantity_type>::format(v.common(), ctx);
-  }
-};
-
 namespace glide_computer {
 
-template<units::QuantityKind QK1, units::QuantityKind QK2>
-constexpr units::Dimensionless auto operator/(const QK1& lhs, const QK2& rhs)
-  requires(!units::QuantityKindRelatedTo<QK1, QK2>) && requires { lhs.common() / rhs.common(); }
-{
-  return lhs.common() / rhs.common();
-}
-
-// kinds
-using horizontal_kind = geographic::horizontal_kind;
-struct vertical_kind : units::kind<vertical_kind, units::isq::si::dim_length> {};
-struct vertical_point_kind : units::point_kind<vertical_point_kind, vertical_kind> {};
-struct velocity_kind : units::derived_kind<velocity_kind, units::isq::si::dim_speed, horizontal_kind> {};
-struct rate_of_climb_kind : units::derived_kind<rate_of_climb_kind, units::isq::si::dim_speed, vertical_kind> {};
-
 // https://en.wikipedia.org/wiki/Flight_planning#Units_of_measurement
+inline constexpr struct mean_sea_level : units::absolute_point_origin<units::isq::height> {
+} mean_sea_level;
+QUANTITY_SPEC(rate_of_climb_speed, units::isq::height / units::isq::time);
+
 // length
-using distance = units::quantity_kind<horizontal_kind, units::isq::si::kilometre>;
-using height = units::quantity_kind<vertical_kind, units::isq::si::metre>;
-using altitude = units::quantity_point_kind<vertical_point_kind, units::isq::si::metre>;
+using distance = units::quantity<units::isq::distance[units::si::kilo<units::si::metre>]>;
+using height = units::quantity<units::isq::height[units::si::metre]>;
+using altitude = units::quantity_point<units::isq::altitude[units::si::metre], mean_sea_level>;
 
 // time
-using duration = units::isq::si::time<units::isq::si::second>;
-using timestamp = units::quantity_point<units::clock_origin<std::chrono::system_clock>, units::isq::si::second>;
+using duration = units::quantity<units::isq::duration[units::si::second]>;
+using timestamp =
+  units::quantity_point<units::isq::time[units::si::second], units::chrono_point_origin<std::chrono::system_clock>{}>;
 
 // speed
-using velocity = units::quantity_kind<velocity_kind, units::isq::si::kilometre_per_hour>;
-using rate_of_climb = units::quantity_kind<rate_of_climb_kind, units::isq::si::metre_per_second>;
+using velocity = units::quantity<units::isq::speed[units::si::kilo<units::si::metre> / units::si::hour]>;
```

Comment:
I am not sure if that is a good idea. Not all units will be in the SI namespace. And I am not talking about the user-defined ones but also the ones provided with the library (i.e. iec80000). Additionally, I chose a variable template for similarity. `kilo<metre>` produces `kilo_<metre>` type visible in the compiler error. I also hope that at some point, it will be allowed to name templates the same as objects (if I am not wrong, a paper about it is coming), so in this case, both will be exactly the same.

Reply:
> Additionally, I chose a variable template for similarity. `kilo<metre>` produces `kilo_<metre>` type visible in the compiler error.

`kilo(metre)` would still return that type.

> I also hope that at some point, it will be allowed to name templates the same as objects (if I am not wrong, a paper about it is coming), so in this case, both will be exactly the same.

Given that
```C++
struct x;
void x();
```
also works today, I imagine such a proposal would also make it work of the type were a template, i.e. whether the other entity is a variable or a function is irrelevant.

> Not all units will be in the SI namespace. And I am not talking about the user-defined ones but also the ones provided with the library (i.e. iec80000).

How about moving the prefixes to the library's top-level namespace? The SI and binary prefixes are more universal than the other entities they share a namespace with.

Reply:
As you can see at <https://github.com/mpusz/units/pull/350/files#diff-c45a6adabc811ade80459a55314772e4054cdcf7804228bb714231f9da5ec7eaR49> and <https://github.com/mpusz/units/pull/350/files#diff-49ddb92142bdd1a382bf5ffe66cc5a2a71b7f1f17d69e328cabacee221740677>,
I had to move the prefixes to `core`.

Reply:
I will think about it. Officially we have prefixes defined by the SI and IEC8000 documents, and our namespaces reflect that now. Moving both to the core would lose that separation but will make it shorter to type. I am wondering if the modularization of V2 will require the same approach?

Reply:
I only moved them.
I didn't rename them nor changed their namespace.
I'm OK with those staying the same.
It's just that their use shouldn't require importing all of SI.